include/deal.II/grid/grid_tools.h

```diff
@@ -983,6 +987,73 @@ namespace GridTools
   get_patch_around_cell(const typename Container::active_cell_iterator &cell);
 
 
+  /**
+   * This function takes a vector of active cells (here named as patch_cells) 
+   * as input argument, and returns vector of their parent cells with the 
+   * coarsest common level of refinement.
+   * The way that the function works, is as follows:
+   * first it computes the minimum refinement level of the given vector of
+   * active cells. Then loops over cells of the input vector of active cells.
+   * For each cell, if its refinement level is equal to the computted minimum 
+   * refinement level, that cell is pushed back to the vector of output cells.
+   * Otherwise, the function looks for the parent cell of the current cell, 
+   * with the level equal to the computed minimum refinement level, and 
+   * then pushes back this parent cell to the vector of output cells.
+   *
+   * @tparam Container
+   * In C++, the compiler can not determine the type of <code>Container</code>
+   * from the function call. You need to specify it as an explicit template
+   * argument following the function name.
+   * @param[in] patch_cells This vector of cells, typically results from 
+   * calling the function GridTools::get_patch_around_cell().
+   * @return A list of cells with the coarsest common level of refinement
+   *
+   * @author Arezou Ghesmati, Wolfgang Bangerth, 2015
+   */
+  template <class Container>
+  std::vector<typename Container::cell_iterator> get_cells_at_coarsest_common_level(
+    const std::vector<typename Container::active_cell_iterator> &patch_cells);
+
+  /**
+   * The first element of input argument is a vector of active cells which we
```

Comment:
1) patch_cells could be any vector of active_cells...
 Yes, there are other active cells in the local_triangulation.
However, as the code shows by flagging the ones which exist in the patch, using "set_user_flag", we ensure which cells in the local_triangulation are flagged for future use... so that we will know which cells are the ones which matters to us.
For example, in my hp-FEM code, when I want to solve the local variational problem on these local_triangulations, I used "user_flag_set==True/False"  to see which cell should be assigned active_fe_index/FE_Nothing, respectively.

2) No, the parent cells are deleted from the map. Instead, their children are added.